File: rotatedBinarySearchTarget.js
function rotatedBinarySearchTarget(nums, target) {
	let l = 0, r = nums.length - 1

	while (l <= r) {
		const mid = Math.floor((l + r) / 2)

		if (nums[mid] === target) return mid

		if (nums[l] <= nums[mid]) {
			if (nums[l] > target || target > nums[mid])
				l = mid + 1
			else 
				r = mid - 1
		} else {
			if (nums[r] < target || target < nums[mid])
				r = mid - 1
			else 
				l = mid + 1
		}
	}

	return -1
}

console.log(rotatedBinarySearchTarget([4, 5, 6, 1, 2, 3], 5))

File: rotatedBinarySearchTarget.cpp
// not working
#include "iostream"
#include "conio"
using namespace std;

int rotatedBinarySearchTarget(int nums[], int target) {
	int l = 0;
	// int length = sizeof(nums)/sizeof(nums[0]);
	int length = *(&nums + 1) - nums;
	int r = length - 1;
	// cout<<r;
	return r;
}

int main(int argc, char const *argv[])
{
	clrscr();
	int nums[] = {6, 7, 8, 1, 2, 3};
	cout<<rotatedBinarySearchTarget(nums, 2)<<endl;
	return 0;
}